source/opt/eliminate_dead_constant_pass.cpp

```diff
@@ -96,25 +95,23 @@ Pass::Status EliminateDeadConstantPass::Process(ir::IRContext* irContext) {
   // constants.
   std::unordered_set<ir::Instruction*> dead_others;
   for (auto* dc : dead_consts) {
-    if (analysis::UseList* uses =
-            irContext->get_def_use_mgr()->GetUses(dc->result_id())) {
-      for (const auto& u : *uses) {
-        if (ir::IsAnnotationInst(u.inst->opcode()) ||
-            ir::IsDebug1Inst(u.inst->opcode()) ||
-            ir::IsDebug2Inst(u.inst->opcode()) ||
-            ir::IsDebug3Inst(u.inst->opcode())) {
-          dead_others.insert(u.inst);
-        }
+    irContext->get_def_use_mgr()->ForEachUser(dc, [&dead_others](ir::Instruction* user) {
+      SpvOp op = user->opcode();
+      if (ir::IsAnnotationInst(op) ||
+          ir::IsDebug1Inst(op) ||
+          ir::IsDebug2Inst(op) ||
+          ir::IsDebug3Inst(op)) {
+        dead_others.insert(user);
```

Comment:
KillDef below not deletes the names and decorations, so we can probably delete everything related to ```dead_others```

Reply:
So true.  Good catch.  (Removing "not" from your sentence first..)

Or even invoke KillInst on each member of dead_consts rather than KillDef.

Agree that is good for a future cleanup.